### PR TITLE
Add logging functionality

### PR DESCRIPTION
**Pull request by f321x**
Fixes: Issue https://github.com/f321x/cashu-escrow-kit/issues/13

Added the log crate to have the logging macros available and the env_logger crate to print logs to the stdout.
For critical errors i used error!
For unpleasant but acceptable behavior warn!
For information relevant to the end user info!
For debugging information containing runtime specific information debug!

Used #[allow(unused_imports)] for the log import to always have all log macros available without compiler warnings, it's nicer to have them on hand and we'll probably need most of them anyways at one point.
And for simple traces who only indicate their location trace!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,8 @@ members = [
     "common",
 ]
 
+resolver = "2"
```
Comment (f321x) [inline Cargo.toml:9]: This not being explicitly specified caused the warning:
```
[nix-shell:~/code/birdseed-escrow/client]$ cargo run
warning: virtual workspace defaulting to `resolver = "1"` despite one or more workspace members being on edition 2021 which implies `resolver = "2"`
note: to keep the current resolver, specify `workspace.resolver = "1"` in the workspace root's manifest
note: to use the edition 2021 resolver, specify `workspace.resolver = "2"` in the workspace root's manifest
note: for more details see https://doc.rust-lang.org/cargo/reference/resolver.html#resolver-versions
```
Version 2 is the correct (newer) one

File: client/src/escrow_client/mod.rs [hunk 2]
```diff
@@ -128,7 +128,7 @@ impl RegisteredEscrowClient {
         let wallet = &self.ecash_wallet;
 
         let message = self.nostr_client.receive_escrow_message(20).await?;
-        dbg!("Received Token, vaidating it...");
+        trace!("Received Token, vaidating it...");
```
Comment (rodant) [inline client/src/escrow_client/mod.rs:131]: Typo: validating
```suggestion
        trace!("Received Token, validating it...");
```

File: client/src/escrow_client/mod.rs [hunk 3]
```diff
@@ -151,10 +151,10 @@ impl TokenExchangedEscrowClient {
         // todo: as buyer either send signature or begin dispute
         match self.trade_mode {
             TradeMode::Buyer => {
-                dbg!("Payed invoince and waiting for delivery...");
+                trace!("Payed invoince and waiting for delivery...");
```
Comment (rodant) [inline client/src/escrow_client/mod.rs:154]: Typo.
```suggestion
                trace!("Payed invoice and waiting for delivery...");
```

File: client/src/escrow_client/mod.rs [hunk 4]
```diff
@@ -151,10 +151,10 @@ impl TokenExchangedEscrowClient {
...
             }
             TradeMode::Seller => {
-                dbg!("Got payment and proceding with delivery...");
+                trace!("Got payment and proceding with delivery...");
```
Comment (rodant) [inline client/src/escrow_client/mod.rs:157]: Typo.
```suggestion
                trace!("Got payment and proceeding with delivery...");
```

**Review by rodant — APPROVED** [review]

**Pull request merged**